models/zeroex/arbitrum/zeroex_arbitrum_native_fills.sql

```diff
@@ -0,0 +1,240 @@
+                  WHEN tp.symbol = 'DAI' THEN (fills.takerTokenFilledAmount / 1e18) * tp.price
+                  WHEN mp.symbol = 'DAI' THEN (fills.makerTokenFilledAmount / 1e18) * mp.price
+                  WHEN tp.symbol = 'WETH' THEN (fills.takerTokenFilledAmount / 1e18) * tp.price
+                  WHEN mp.symbol = 'WETH' THEN (fills.makerTokenFilledAmount / 1e18) * mp.price
+                  ELSE COALESCE((fills.makerTokenFilledAmount / (10^mt.decimals))*mp.price,(fills.takerTokenFilledAmount / (10^tt.decimals))*tp.price)
+              END AS volume_usd
+          , NULL::NUMERIC AS protocol_fee_paid_eth
+      FROM {{ source('zeroex_arbitrum', 'ExchangeProxy_evt_RfqOrderFilled') }} fills
+      LEFT JOIN prices.usd tp ON
+          date_trunc('minute', evt_block_time) = tp.minute 
+          AND CASE
+                  -- set native token to wrapped version
+                    WHEN fills.takerToken = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' THEN '0x82af49447d8a07e3bd95bd0d56f35241523fbab1'
+                    ELSE fills.takerToken
+              END = tp.contract_address
+      LEFT JOIN prices.usd mp ON
+          DATE_TRUNC('minute', evt_block_time) = mp.minute  
+          AND CASE
+                  -- set native token to wrapped version
+                    WHEN fills.makerToken = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' THEN '0x82af49447d8a07e3bd95bd0d56f35241523fbab1'
+                    ELSE fills.makerToken
+              END = mp.contract_address
+      LEFT OUTER JOIN {{ ref('tokens_erc20') }} mt ON mt.contract_address = fills.makerToken
+      LEFT OUTER JOIN {{ ref('tokens_erc20') }} tt ON tt.contract_address = fills.takerToken
+       where 1=1  and  mp.blockchain = 'arbitrum' and tp.blockchain = 'arbitrum'
+                {% if is_incremental() %}
+                AND evt_block_time >= date_trunc('day', now() - interval '1 week')
+                {% endif %}
+                {% if not is_incremental() %}
+                AND evt_block_time >= '{{zeroex_v3_start_date}}'
+                {% endif %}
+    ), otc_fills as
+    (
+      SELECT
+          fills.evt_block_time AS block_time
+          , 'otc' as native_order_type
+          , 'v4' AS protocol_version
+          , fills.evt_tx_hash AS transaction_hash
+          , fills.evt_index
+          , fills.maker AS maker_address
+          , fills.taker AS taker_address
+          , fills.makerToken AS maker_token
+          , fills.takerTokenFilledAmount as taker_token_filled_amount_raw
+          , fills.makerTokenFilledAmount as maker_token_filled_amount_raw
+          , fills.contract_address 
+          , mt.symbol AS maker_symbol
+          , fills.makerTokenFilledAmount / (10^mt.decimals) AS maker_asset_filled_amount
+          , fills.takerToken AS taker_token
+          , tt.symbol AS taker_symbol
+          , fills.takerTokenFilledAmount / (10^tt.decimals) AS taker_asset_filled_amount
+          , NULL: AS fee_recipient_address
```

Comment:
can you `cast()` null to the datatype? while you have an assigned value in the prior CTE, and datatype should be inferred already, it's safe to apply here